components/list/list.js

```diff
@@ -207,6 +218,9 @@ class List extends PageableMixin(SelectionMixin(LitElement)) {
 		if (changedProperties.has('breakpoints') && changedProperties.get('breakpoints') !== undefined) {
 			this.resizedCallback(this.offsetWidth, true);
 		}
+		if (changedProperties.has('add-button') || changedProperties.has('add-button-text')) {
+			this._listChildrenUpdatedSubscribers.updateSubscribers();
```

Comment:
I tabbed into the updating subscribers code in order to update the list children when `add-button` or `add-button-text` get changed. Not sure if this is totally ideal, as it also runs the other update functions. If we don't think this is going to change post-initial-render, it could also be done like `grid` and `extend-separators` where those get set for the child list items in `list-item-role-mixin`

Reply:
I think the way `ListItemRoleMixin` works, it only checks for `grid` and `extend-separators` in the `connectedCallback`. I think there is an assumption (which is probably fairly safe), that those won't change once the item is connected to a list.

For this case though, might we ever expect the consumer to change `addButton` or `addButtonText` after the items have rendered, perhaps in response to some user action? I'm honestly not sure, but your current solution would appear to handle that case.

Reply:
It's totally possible it could change in response to a user action, so it sounds like this way might be the better option.